pkg/app/redis.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2019 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/helm"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/log"
+	"github.com/kanisterio/kanister/pkg/poll"
+	"github.com/pkg/errors"
+)
+
+const (
+	redisWaitTimeout = 1 * time.Minute
+)
+
+type RedisDB struct {
+	cli       kubernetes.Interface
+	namespace string
+	name      string
+	chart     helm.ChartInfo
+}
+
+// Last tested working version "7.0.7"
+func NewRedisDB(name string) App {
+	return &RedisDB{
+		name: name,
+		chart: helm.ChartInfo{
+			Release:  appendRandString(name),
+			RepoURL:  helm.BitnamiRepoURL,
+			Chart:    "redis",
+			RepoName: helm.BitnamiRepoName,
+			Values: map[string]string{
+				"auth.rootPassword":         "mysecretpassword",
+				"image.pullPolicy":          "Always",
+				"volumePermissions.enabled": "true",
+				"master.disableCommands":    "",
+			},
+		},
+	}
+}
+
+func (rdb *RedisDB) Init(ctx context.Context) error {
+	cfg, err := kube.LoadConfig()
+	if err != nil {
+		return err
+	}
+	rdb.cli, err = kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (rdb *RedisDB) Install(ctx context.Context, namespace string) error {
+	rdb.namespace = namespace
+	cli, err := helm.NewCliClient()
+	if err != nil {
+		return errors.Wrap(err, "failed to create helm client")
+	}
+	log.Print("Adding repo.", field.M{"app": rdb.name})
+	err = cli.AddRepo(ctx, rdb.chart.RepoName, rdb.chart.RepoURL)
+	if err != nil {
+		return errors.Wrapf(err, "Error adding helm repo for app %s.", rdb.name)
+	}
+	log.Print("Installing redis instance using helm.", field.M{"app": rdb.name})
+	err = cli.Install(ctx, rdb.chart.RepoName+"/"+rdb.chart.Chart, rdb.chart.Version, rdb.chart.Release, rdb.namespace, rdb.chart.Values, true)
+	if err != nil {
+		return errors.Wrapf(err, "Error intalling application %s through helm.", rdb.name)
+	}
+	return nil
```

Comment:
```suggestion
	return errors.Wrapf(err, "Error intalling application %s through helm.", rdb.name)
```
`errors.Wrapf()` returns nil if the err is nil https://pkg.go.dev/github.com/pkg/errors#Wrapf

Reply:
fixed